src/CommonDBTM.php

```diff
@@ -1185,6 +1186,21 @@ protected function restoreInput(array $default = [])
         return $default;
     }
 
+    public function callFormAction(string $form_action, SymfonyRequest $request): mixed
+    {
+        $post_data = $request->request->all();
+
+        // POST action execution
+        return match ($form_action) {
+            'add' => $this->add($post_data),
+            'delete' => $this->delete($post_data),
+            'restore' => $this->restore($post_data),
+            'purge' => $this->delete($post_data, 1),
+            'update' => $this->update($post_data),
+            default => throw new \RuntimeException(\sprintf("Unsupported object action \"%s\".", $form_action->value)),
+        };
+    }
```

Comment:
IMHO, it should not be possible to override this. If an item requires a specifiic action to be implemented, then it should probably be done in a specific controller, like you did in the `ContactFormController`.

Reply:
This is mandatory in case you need custom checks to validate whether a specific form action can be executed, or when you need to call some code *after* some action is executed. So far, there is no custom way to do this unless creating a custom controller, which I think is overkill for some cases, though viable in other situations.